src/main/java/net/barrage/school/java/ecatalog/app/ProductServiceImpl.java

```diff
@@ -1,35 +1,32 @@
 package net.barrage.school.java.ecatalog.app;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import lombok.Data;
-import lombok.RequiredArgsConstructor;
 import lombok.SneakyThrows;
 import net.barrage.school.java.ecatalog.model.Product;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
-import java.io.File;
 import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
-import java.util.Optional;
-import java.util.UUID;
 
-@RequiredArgsConstructor
 @Service
 public class ProductServiceImpl implements ProductService {
 
-    // FYI - https://www.baeldung.com/jackson-object-mapper-tutorial
-    private final ObjectMapper objectMapper;
+    private final List<ProductSource> productSources;
 
-    @Value("${ecatalog.products.source-path}")
-    private File productsSourceFile;
+    public ProductServiceImpl(
+            List<ProductSource> productSources) {
+        this.productSources = productSources;
+    }
 
     @SneakyThrows
     @Override
     public List<Product> listProducts() {
-        return objectMapper.readValue(productsSourceFile, SourceProductList.class).stream()
-                .map(sourceProduct -> convert(sourceProduct))
-                .toList();
+        List<Product> products = new ArrayList<Product>();
+        for (ProductSource source : productSources) {
+            products.addAll(source.getProducts());
+        }
+        products.sort(Comparator.comparing(Product::getName));
+        return products;
     }
 
     @SneakyThrows
```

Comment:
Why do u need `@SneakyThrows` here?

Reply:
Forgot to remove them when updating functions